.github/workflows/biotools-gh2biotools.yml

```diff
@@ -0,0 +1,123 @@
+        id: check_changes
+        run: |
+          files=$(git diff-tree --no-commit-id --name-only --diff-filter=d -r ${{ github.sha }} | grep '\.biotools\.json$' || true)
+
+          if [ -z "$files" ]; then
+            echo "No relevant biotools.json files changed."
+            echo "changed=false" >> $GITHUB_OUTPUT
+            exit 0
+          else
+            echo "Files changed:"
+            echo "$files"
+            echo "changed=true" >> $GITHUB_OUTPUT
+            echo "files=${files}" >> $GITHUB_OUTPUT
+          fi
+
+
+      - name: Validate allowed fields in modified biotools.json files
+        if: steps.check_changes.outputs.changed == 'true'
+        run: |
+          echo "Validating fields in changed .biotools.json files..."
+          PROTECTED_FIELDS=(
+            "biotoolsID"
+            "biotoolsCURIE"
+            "additionDate"
+            "collectionID"
+            "elixirPlatform"
+            "elixirNode"
+            "elixirCommunity"
+            "lastUpdate"
+            "owner"
+            "editPermission"
+            "validated"
+            "homepage_status"
+            "elixir_badge"
+            "confidence_flag"
+          )
+
+          failed=false
+
+          for file in $(echo "${{ steps.check_changes.outputs.files }}"); do
+            echo "Checking $file..."
+        
+            git show ${{ github.event.before }}:$file > old.json || continue
+            cat $file > new.json
+
+            for field in "${PROTECTED_FIELDS[@]}"; do
+              old_val=$(jq -e ".$field" old.json 2>/dev/null || echo "")
+              new_val=$(jq -e ".$field" new.json 2>/dev/null || echo "")
+              if [ "$old_val" != "$new_val" ]; then
+                echo "❌ Field '$field' has been modified in $file"
+                failed=true
+              fi
+            done
+          done
+
+          if [ "$failed" = true ]; then
+            echo "::error ::One or more not allowed fields have been modified."
+            exit 1
+          else
+            echo "✅ All changed fields are allowed."
+          fi
+
+
+      - name: Checkout utils repo
+        if: steps.check_changes.outputs.changed == 'true'
+        uses: actions/checkout@v3
+        with:
+          repository: research-software-ecosystem/utils
+          path: content-ecosystem-utils
+          
+
+      - name: Install dependencies
+        if: steps.check_changes.outputs.changed == 'true'
+        working-directory: content-ecosystem-utils/scripts/runbiotools
+        run: |
+          pip3 install -r requirements.txt
+
+
+      - name: Run gh2biotools.py script
+        if: steps.check_changes.outputs.changed == 'true'
+        env:
+          BIOTOOLS_API_TOKEN: ${{ secrets.BIOTOOLS_API_TOKEN }}
+        run: |
+          changed_files="${{ steps.check_changes.outputs.files }}"
+
+          command="python content-ecosystem-utils/scripts/runbiotools/gh2biotools.py --token \"$BIOTOOLS_API_TOKEN\""
```

Comment:
if the API token is not printed on the command line then it will not be printed in the logs, which might be better security-wise. I know that to some extent github "hides" secrets in the logs, but would it be possible just to rather read directly the token from the environment in the python script?